pkg/distributor/query.go

```diff
@@ -196,15 +199,33 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 		resp.Chunkseries = append(resp.Chunkseries, series)
 	}
 	for _, series := range hashToTimeSeries {
-		sort.Sort(byTimestamp(series.Samples))
 		resp.Timeseries = append(resp.Timeseries, series)
 	}
 
 	return resp, nil
 }
 
-type byTimestamp []client.Sample
+// Merges and dedupes two sorted slices with samples together, storing result in the first slice.
+// If a and b overlap, weird things will happen.
```

Comment:
What does `if a and b overlap` mean? If we're deduping, then they're overlapping?

I had to read the code to understand that you meant the slice addresses overlapping, but best to make it clear in the comment?

Reply:
I've modified the code, and removed the comment.